src/System.Linq/tests/AllTests.cs

```diff
@@ -14,90 +13,58 @@ public class AllTests : EnumerableTests
 
-            Func<string, bool> predicate = String.IsNullOrEmpty;
-
+            Func<string, bool> predicate = string.IsNullOrEmpty;
             Assert.Equal(q.All(predicate), q.All(predicate));
         }
 
-        [Fact]
-        public void SourceIsEmpty()
-        {
-            int[] source = { };
-
-            Assert.True(source.All(IsEven));
-        }
-
-        [Fact]
-        public void OneElementPredicateFalse()
-        {
-            int[] source = { 3 };
-
-            Assert.False(source.All(IsEven));
-        }
-
-        [Fact]
-        public void OneElementPredicateTrue()
-        {
-            int[] source = { 4 };
-
-            Assert.True(source.All(IsEven));
-        }
-
-        [Fact]
-        public void PredicateTrueOnSomeButNotAll()
+        public static IEnumerable<object[]> All_TestData()
         {
-            int[] source = { 4, 8, 3, 5, 10, 20, 12 };
-
-            Assert.False(source.All(IsEven));
+            Func<int, bool> isEvenFunc = IsEven;
+            yield return new object[] { new int[0], isEvenFunc, true };
+            yield return new object[] { new int[] { 3 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4 }, isEvenFunc, true };
+            yield return new object[] { new int[] { 3 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4, 8, 3, 5, 10, 20, 12 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4, 2, 10, 12, 8, 6, 3 }, isEvenFunc, false };
+            yield return new object[] { new int[] { 4, 2, 10, 12, 8, 6, 14 }, isEvenFunc, true };
+
+            int[] range = Enumerable.Range(1, 10).ToArray();
+            yield return new object[] { range, (Func<int, bool>)(i => i > 0), true };
+            for (int j = 1; j <= 10; j++)
+            {
+                int k = j; // Local copy for iterator
+                yield return new object[] { range, (Func<int, bool>)(i => i > k), false };
+            }
         }
 
-        [Fact]
-        public void PredicateTrueAllExceptLast()
-        {
-            int[] source = { 4, 2, 10, 12, 8, 6, 3 };
-
-            Assert.False(source.All(IsEven));
-        }
-
-        [Fact]
-        public void TrueForAll()
-        {
-            int[] source = { 4, 2, 10, 12, 8, 6, 14 };
-
-            Assert.True(source.All(IsEven));
-        }
-
-        [Fact]
-        public void RangeAllWithinRange()
+        [Theory]
+        [MemberData(nameof(All_TestData))]
+        public void All(IEnumerable<int> source, Func<int, bool> predicate, bool expected)
         {
-            var array = Enumerable.Range(1, 10).ToArray();
-            Assert.True(array.All(i => i > 0));
-            for (var j = 1; j <= 10; j++)
-                Assert.False(array.All(i => i > j));
+            Assert.Equal(expected, source.All(predicate));
         }
 
         [Fact]
-        public void NullSource()
+        public void All_NullSource_ThrowsArgumentNullException()
```

Comment:
We're already in the AllTests class... why do we need an All prefix on the tests?